daemon/cmd/kube_proxy_replacement.go

```diff
@@ -155,6 +155,13 @@ func initKubeProxyReplacementOptions() error {
 			return fmt.Errorf("Invalid value for --%s: %s", option.NodePortAcceleration, option.Config.NodePortAcceleration)
 		}
 
+		if option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled &&
+			option.Config.EnableWireguard && option.Config.EncryptNode {
+			log.Warnf("With %s: %s and %s, %s enabled, N/S Loadbalancer traffic won't be encrypted "+
```

Comment:
@brb and I talked about this. @brb What do you think?
https://github.com/cilium/cilium/issues/25354#issuecomment-1552473562

> I think for now we could add a warning to the agent saying that N/S LB traffic won't be encrypted, and then protect the invocation of wg_maybe_redirect_to_encrypt() in encap.h with #if __ctx_is == __ctx_skb.

Reply:
At the moment, `--encrypt-node` is used to encrypt both - N/S LB traffic and node-to-node traffic (incl. node-to-pod and pod-to-node). If we fatal, then the latter with the XDP on won't be possible.

I think the cleanest solution would be to introduce `--encrypt-lb`, and then panic if it's enabled with the XDP. @gandro WDYT?

Reply:
Given these limitations, yeah I think an `--encrypt-lb` makes sense.

Reply:
`--encrypt-lb` is true, and either XDP or DSR is enabled, then we throw the fatal error? The DSR doesn't support N/S LB traffic encryption, right?

I also have a feeling that SNAT mode + Tunnel  + No XDP is also not supported
https://github.com/cilium/cilium/issues/25354#issuecomment-1565456915

Reply:
> --encrypt-lb is true, and either XDP or DSR is enabled, then we throw the fatal error? The DSR doesn't support N/S LB traffic encryption, right?

Yep. I can introduce the flag, or let me know if you want to do this.

> I have a feeling that SNAT mode + Tunnel + No XDP is also not supported

Yep, it's listed in https://github.com/cilium/cilium/issues/23032.

Reply:
> let me know if you want to do this.

I want to do this, but I can't test the changes to conformance-e2e. Is there any way to do it?
> It will not work with a branch from a fork (missing secrets).
https://github.com/cilium/cilium/blob/7eb40a095a5407b90afc3c489b6089c83db582f3/.github/workflows/conformance-e2e.yaml#L11-L24

We need to add `--encrypt-lb` to conformance-e2e, right?
https://github.com/cilium/cilium/pull/19401/files#diff-9e53f731e74bddbc0167de35296f2ae4325ca6ea679cc2bbd21635f3e44e7c31

Reply:
Also, we need to consider a way to opt out of encrypting LB traffic. One possible solution is to mark packets to skip encryption at nodeport.h.

Reply:
> I want to do this, but I can't test the changes to conformance-e2e. Is there any way to do it?

https://github.com/cilium/cilium/pull/25813#issuecomment-1579934285

> We need to add --encrypt-lb to conformance-e2e, right?

Only to the configurations which run with VXLAN + WG. We don't have LB XDP yet (I will add it).

> One possible solution is to mark packets to skip encryption at nodeport.h.

I think an SKB mark is a scarce resource, so let's better add if-defs :-)

Reply:
How can we distinguish LB traffic and node-to-node traffic here?
https://github.com/cilium/cilium/blob/2ec713e8e3c94ea26aae4dae8d3b359b633c53ab/bpf/lib/wireguard.h#L86-L89

Reply:
Aye, you are right. At the moment, it's impossible to distinguish between LB and non-LB traffic in the linked code. We can use `skb->tc_index` for that (we already to in some places, just grep for it). Basically, set it in `nodeport_lb{4,6}()` when a backend is remote, and then in `wireguard.h` bypass the encryption if `--encrypt-lb=false` and the corresponding bit is set in the `tc_index`.

Reply:
Thanks! I will look into that

Reply:
We found out that more time is needed to introduce `encrypt-lb` flag to opt-out N/S traffic encryption completely. 
https://github.com/cilium/cilium/pull/26095#issuecomment-1588369930

Shall we proceed with this PR as it is? Or should we make changes to fatal the agent when either XDP or DSR is enabled?

Reply:
After back-and-forth - let's not panic. Instead, document the limitation as this PR already does, and then log a message to users. This will allow XDP users to encrypt node-to-node traffic.

Reply:
Looks good to me! We have a soft rule that every warning should be something where the user can do something to make the warning go away. Should we mention in the warning message that to encrypt that LB traffic, one has to disable XDP? 

We have this "hint" system in other places too, see e.g. 
https://github.com/cilium/cilium/blob/8735ad61f2c2f41faa589174c5d9586b7b49dcad/pkg/wireguard/agent/agent.go#L129-L132

Reply:
Thank you for your comment! Is it something like this?

```go
log.WithField(logfields.Hint,
	"Disable XDP acceleration to encrypt N/S Loadbalancer traffic.").
	Warnf("With %s: %s and %s, %s enabled, N/S Loadbalancer traffic won't be encrypted "+
		"when an intermediate node redirects a request to another node where a selected backend is running.",
		option.NodePortAcceleration, option.Config.NodePortAcceleration, option.EnableWireguard, option.EncryptNode)
```